packages/shared-ux/button_toolbar/src/toolbar/toolbar.tsx

```diff
@@ -8,19 +8,19 @@
 
 import React, { ReactElement } from 'react';
 import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
-
 import { i18n } from '@kbn/i18n';
-import { IconButtonGroup, PrimaryButton } from '../buttons';
+
 import { ToolbarPopover } from '../popover';
+import { IconButtonGroup, ToolbarButton } from '../buttons';
 
 /** type for cases with both button or a popover could be used */
-export type ToolbarButton = typeof PrimaryButton | typeof ToolbarPopover;
+export type ToolbarButtonType = typeof ToolbarButton | typeof ToolbarPopover;
 
 /** Specific type for the toolbar children in its props */
 interface NamedSlots {
-  primaryButton: ReactElement<ToolbarButton>;
+  primaryButton: ReactElement<ToolbarButtonType>;
   iconButtonGroup?: ReactElement<typeof IconButtonGroup>;
-  extraButtons?: Array<ReactElement<ToolbarButton>> | undefined;
+  extraButtons?: Array<ReactElement<ToolbarButtonType>> | undefined;
```

Comment:
Right now, it is expected that the consumer provides `type="primary"` for the `primaryButton` child, like so:

```typescript
      <Toolbar>
        {{
          primaryButton: (
            <ToolbarButton
              type="primary"
              ...
            />
          ),
          iconButtonGroup: ...,
          extraButtons: ...
        }}
      </Toolbar>
```
It is possible we could change this so that the toolbar **itself** is in control of the `type` property - so, it would force `type="primary"` for the `primaryButton` child and it would force `type="empty"` for all of the items in the `extraButtons` child. 

This wasn't an established pattern for other Shared UX components that I could find, so I wasn't sure how hard-handed we wanted to be for the `Toolbar` consumers. Just throwing it out there as an idea!